### PR TITLE
TPCClusterFinder: Some code cleanup

### PR DESCRIPTION
**Pull request by fweig**
This PR includes some cleanup for the TPCClusterFinder:
* Replace `size_t` with 4 byte integers where possible.
* Simplify kernel names.
* Cleanup macros.
* Remove old, obsolete implementation.

### CODE REVIEW

File: GPU/Common/GPUDefGPUParameters.h [hunk 1]
```diff
@@ -285,15 +285,15 @@
...
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanUpStart GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanUp GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanTop GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanDown GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_compact GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step0 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step1 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step2 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step3 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step4 GPUCA_THREAD_COUNT_SCAN
```
Comment (davidrohr) [inline GPU/Common/GPUDefGPUParameters.h:296]: Actually, why do you change this from the names to step[n] and then to integers below?
In my opinion, it was easier to read before.

File: GPU/GPUTracking/TPCClusterFinder/CfUtils.h [hunk 2]
```diff
@@ -77,8 +81,9 @@ class CfUtils
     ushort x = ll % N;
     ushort y = ll / N;
     tpccf::Delta2 d = neighbors[x + offset];
-    LOOP_UNROLL_ATTR for (unsigned int i = y; i < wgSize; i += (elems / N))
-    {
+
+    GPUCA_UNROLL(, U())
+    for (unsigned int i = y; i < wgSize; i += (elems / N)) {
```
Comment (davidrohr) [inline GPU/GPUTracking/TPCClusterFinder/CfUtils.h:86]: If you want to unroll on AMD and NVIDIA, you should do GPUCA_UNROLL(U(), U())

File: GPU/Common/GPUDefGPUParameters.h (hunk 1 above)
Comment (fweig) [reply GPU/Common/GPUDefGPUParameters.h:296]: I thought it looked a bit cleaner. But it does hurt readability. I'll revert it.

**Comment by fweig** [comment]
Looks like the CI didn't like the unrolling of that loop anyway. I think there are more places in the cluster finder where loop unrolling might be beneficial. But that's for another PR.

Fixed the naming of the kernels with the last commit and fixed a minor bug in the noise suppression where the wrong neighbors were checked for minima. I'll work on the gather bug now.

**Comment by davidrohr** [comment]
@fweig : Unfortunately, this seems to break something. Running with my dataset o2-100 on asustest04, the clusterizer crashes after this PR, and it works again after I revert it. Can you have a look?

**Pull request merged**